editor/tests/org.fusesource.ide.camel.editor.tests.integration/src/main/java/org/fusesource/ide/camel/editor/integration/CamelEditorIT.java

```diff
@@ -307,6 +308,85 @@ public void dropRouteOnRoute() throws Exception {
 		assertThat(createRouteFigureFeature.canExecute(createCtx)).isFalse();
 	}
 	
+	@Test
+	public void reconnectToUnwiredNode() throws Exception {
+		IEditorPart openEditorOnFileStore = openFileInEditor("/reconnect.xml");
+		
+		readAndDispatch(20);
+		
+		CamelDesignEditor ed = ((CamelEditor)openEditorOnFileStore).getDesignEditor();
+		IFeatureProvider fp = ed.getFeatureProvider();
+		CamelFile model = ed.getModel();
+		
+		AbstractCamelModelElement route = model.findNode("route1");
+		AbstractCamelModelElement inbox = model.findNode("inbox");
+		
+		CreateContext createCtx = new CreateContext();
+		GraphicsAlgorithm exisitngRoutegraphic = fp.getPictogramElementForBusinessObject(route).getGraphicsAlgorithm();
+		createCtx.setX(exisitngRoutegraphic.getX());
+		createCtx.setY(exisitngRoutegraphic.getY() + exisitngRoutegraphic.getWidth() + 5);
+		createCtx.setTargetContainer((ContainerShape)fp.getPictogramElementForBusinessObject(route));
+		CreateFigureFeature createFigureFeature = new CreateFigureFeature(fp, "Bean", "", CamelModelFactory.getModelForVersion(CamelModelFactory.getLatestCamelVersion()).getEipModel().getEIPByName("bean"));
+		if(createFigureFeature.canExecute(createCtx)){
+			TransactionalEditingDomain editingDomain = CamelUtils.getDiagramEditor().getEditingDomain();
+			CommandExec.getSingleton().executeCommand(new GenericFeatureCommandWithContext(createFigureFeature, createCtx), editingDomain);
+		}
+		
+		AbstractCamelModelElement createdBean = model.findNode("_bean1");
+		assertThat(createdBean).isNotNull();
+
+		// now check if the reconnect is possible - it should be
+		assertThat(ReconnectNodesFeature.canElementsConnect(inbox, createdBean)).isTrue();
+	}
+
+	@Test
+	public void reconnectToWiredNode() throws Exception {

```

Comment:
better to provide more explicit test name: something like: reconnectToWiredNodeShouldNotBepossible